### PR TITLE
Extend Katib Integration Tests

### PR DESCRIPTION
**Pull request by NohaIhab**
Added test to Create an Experiment and make sure it completed entirely using lightkube.

### CODE REVIEW

File: tests/integration/test_charms.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    )
+
+    # Create Experiment instance
+    experiment_file = "examples/v1beta1/hp-tuning/grid-example.yaml"
+    with open(experiment_file) as f:
+        exp = Experiment(yaml.safe_load(f.read()))
+        lightkube_client.create(exp, namespace=namespace)
+
+    # Check if Experiment Succeeded
+    for i in range(30):
```
Comment (DnPlas) [inline tests/integration/test_charms.py:112]: We have done the "retries" using `tenacity`. Please check some of our charms to see how tenacity can help you with retries, [training operator](https://github.com/canonical/training-operator/blob/main/tests/integration/test_charm.py#L81) is one.

File: tests/integration/test_charms.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        experiment = lightkube_client.get(Experiment, name="grid", namespace=namespace)
+
+        try:
+            status = experiment.get("status", {}).get("conditions")[-1].get("type")
+
+        except TypeError:
+            status = None
+
+        if status == "Succeeded":
```
Comment (DnPlas) [inline tests/integration/test_charms.py:122]: You have a `try/except` block above this `if/else` statement. Do you think you could move some of these actions, like the fail and logging messages, inside the `try/except`?

File: tests/integration/test_charms.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            logger.info("Experiment Succeeded")
+            break
+
+        else:
+            logger.info(f"Waiting for Experiment.. Status: {status}")
+
+        time.sleep(5)
+
+    else:
+        pytest.fail("Waited too long for Experiment!")
```
Comment (DnPlas) [inline tests/integration/test_charms.py:132]: In test cases we usually `assert` a result, for example `assert some_var == 1` which will fail or pass your test case depending on the result. Please check pytest's [docs](https://docs.pytest.org/en/7.1.x/how-to/assert.html)  (and even unittest [docs](https://docs.python.org/3/library/unittest.html)) for more info.

File: tests/integration/test_charms.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
+# Copyright 2022 Canonical Ltd.
```
Comment (DnPlas) [inline tests/integration/test_charms.py:1]: What would be the reason you created an `integration` directory?

File: examples/v1beta1/hp-tuning/grid-example.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
+---
+apiVersion: kubeflow.org/v1beta1
+kind: Experiment
+metadata:
+  namespace: kubeflow
```
Comment (DnPlas) [inline examples/v1beta1/hp-tuning/grid-example.yaml:5]: Although there is nothing wrong with defining the namespace, I could see some cases where this is not desired:
1) If users want to apply/create this outside the `kubeflow` namespace
2) If our test environment change and we do not deploy katib in the `kubeflow` namespace anymore

I'd suggest you leave this empty and apply/create the experiment specifying the namespace, which can be anything, like `testing`.

File: tests/integration/test_charms.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+# See LICENSE file for licensing details.
+
+from pathlib import Path
```
Comment (DnPlas) [inline tests/integration/test_charms.py:4]: I suggest you check best practices for python imports so we can improve this import section.

**Review by DnPlas — COMMENTED** [review]
Thanks @NohaIhab, left a couple of comments.

File: tests/integration/test_charms.py (hunk 4 above)
Comment (NohaIhab) [reply tests/integration/test_charms.py:1]: assuming there would be another directory for unit tests

File: tests/integration/test_charms.py [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        version="v1beta1",
+        kind="experiment",
+        plural="experiments",
+        verbs=None,
+    )
+
+    # Create Experiment instance
+    experiment_file = "examples/v1beta1/hp-tuning/grid-example.yaml"
+    with open(experiment_file) as f:
+        exp = Experiment(yaml.safe_load(f.read()))
```
[the remark this answers is not in the document]
Comment (NohaIhab) [reply tests/integration/test_charms.py:108]: class 'lightkube.generic_resource.experiment'

File: examples/v1beta1/hp-tuning/grid-example.yaml (hunk 5 above)
Comment (NohaIhab) [reply examples/v1beta1/hp-tuning/grid-example.yaml:5]: would I then specify the namespace in the tox command?

File: tests/integration/test_charms.py [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+            exp_status == "Running"
+        ), f"{exp_object.metadata.name} not running status = {exp_status})"
+
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(multiplier=2, min=1, max=15),
+        stop=tenacity.stop_after_attempt(10),
+        reraise=True,
+    )
+    def assert_trial_status_running():
+        """Asserts on the trial status.
```
Comment (DnPlas) [inline tests/integration/test_charms.py:171]:
```suggestion
        """Asserts the trial status is Running.
```

**Review by DnPlas — APPROVED** [review]
Thanks @NohaIhab, LGTM!

**Pull request merged**